app/containers/DocumentList/DocumentList.jsx

```diff
@@ -193,25 +209,17 @@ class DocumentList extends React.Component {
     const documentsAreLoading = !data || data.isLoading
 
     if (documentsAreLoading) {
-      // If there is a custom handler for the loading state, we'll call it.
-      if (typeof onLoading === 'function') {
-        return onLoading()
-      }
+      return typeof onLoading === 'function' ? onLoading() : <SpinningWheel />
+    }
 
-      // If not, we render the SpinningWheel component.
-      return <SpinningWheel />
+    if (data.error) {
+      return typeof onError === 'function' ? onError({error: data.error}) : null
     }
 
     const {metadata, results} = data
 
-    // If there are no results to display, we call the `onEmptyList` callback
-    // if it is defined, or render `null` otherwise.
     if (results.length === 0) {
-      if (typeof onEmptyList === 'function') {
-        return onEmptyList({selection})
-      }
-
-      return null
+      return typeof onEmptyList === 'function' ? onEmptyList({selection}) : null
```

Comment:
Seems quicker to read to me, what do you think? Just wanted to do a bit of minor gardening while I'm here